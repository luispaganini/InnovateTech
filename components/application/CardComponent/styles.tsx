import { ThemedView } from '@/components/ThemedView';
import { Animated, ColorSchemeName } from 'react-native';
import styled from 'styled-components/native';
import { Colors } from '@/constants/Colors';

export const Card = styled(ThemedView)<{ colorTheme: ColorSchemeName}>`
    width: 90%;
    height: 100px;
    border-radius: 2px;
    border-width: 1px;
    background-color: ${props => props.colorTheme == 'light' ? Colors.light.background : Colors.dark.background};
    border-color: ${props => props.colorTheme == 'light' ? Colors.light.border : Colors.dark.border};
    justify-content: space-between;
    padding: 10px;
    align-self: center;
    flex-direction: row;
    align-items: center;
    overflow: hidden;
`;

export const CardTextContent = styled(ThemedView)`
    flex: 1;
    padding: 5px;
    margin-left: 10px;
    height: 100%;
    justify-content: space-between;
`;

export const InfoCardText = styled(ThemedView)`
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: wrap;
`;