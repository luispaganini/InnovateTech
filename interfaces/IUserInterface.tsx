export default interface IUserInterface {
    name: string
    lastName: string
    sex: string
    dateOfBirth: Date
}