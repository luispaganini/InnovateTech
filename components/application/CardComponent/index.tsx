import { View, Text, useWindowDimensions } from 'react-native'
import React from 'react'
import { useColorScheme } from '@/hooks/useColorScheme';
import { ThemedView } from '@/components/ThemedView';
import { Avatar } from 'react-native-paper';
import { Card, CardTextContent, InfoCardText } from './styles';
import { ThemedText } from '@/components/ThemedText';
import IUserInterface from '@/interfaces/IUserInterface';
import 'intl-pluralrules';

interface CardComponentProps {
    user: IUserInterface
}


export default function CardComponent(props: CardComponentProps) {
    const colorScheme = useColorScheme();

    const windowWidth = useWindowDimensions().width;
    const avatarSize = windowWidth * 0.2;

    return (
        <Card colorTheme={colorScheme}>
            <Avatar.Image size={avatarSize} source={{ uri: 'https://randomuser.me/api/portraits/men/12.jpg' }} />
            <CardTextContent>
                <ThemedText type="defaultSemiBold">{props.user.name + ' ' + props.user.lastName}</ThemedText>
                <InfoCardText>
                    <ThemedText type="default">{props.user.sex}</ThemedText>
                    <ThemedText type="default">{props.user.dateOfBirth.toLocaleDateString('pt-BR')}</ThemedText>
                </InfoCardText>
            </CardTextContent>
        </Card>
    )
}