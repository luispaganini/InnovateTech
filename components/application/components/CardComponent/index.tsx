import { TouchableOpacity, useWindowDimensions } from 'react-native'
import React from 'react'
import { useColorScheme } from '@/hooks/useColorScheme';
import { Avatar } from 'react-native-paper';
import { Card, CardTextContent, InfoCardText } from './styles';
import { ThemedText } from '@/components/ThemedText';
import IUserInterface from '@/interfaces/IUserInterface';
import 'intl-pluralrules';
import { useTranslation } from 'react-i18next';

interface CardComponentProps {
    user: IUserInterface
    onPress: () => void
}

export default function CardComponent(props: CardComponentProps) {
    const colorScheme = useColorScheme();

    const windowWidth = useWindowDimensions().width;
    const avatarSize = windowWidth * 0.2;
    const { t } = useTranslation();

    return (
        <TouchableOpacity onPress={props.onPress}>
            <Card colorTheme={colorScheme}>
                <Avatar.Image size={avatarSize} source={{ uri: props.user.picture.large }} />
                <CardTextContent>
                    <ThemedText type="defaultSemiBold">{t(props.user.name.title) + ' ' + props.user.name.first + ' ' + props.user.name.last}</ThemedText>
                    <InfoCardText>
                        <ThemedText type="default">{t(props.user.gender)}</ThemedText>
                        <ThemedText type="default">{new Date(props.user.dob.date).toLocaleDateString('pt-BR')}</ThemedText>
                    </InfoCardText>
                </CardTextContent>
            </Card>
        </TouchableOpacity>
    )
}