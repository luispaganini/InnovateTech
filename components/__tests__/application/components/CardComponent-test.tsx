import CardComponent from '@/components/application/components/CardComponent';
import IUserInterface from '@/interfaces/IUserInterface';
import React from 'react';
import renderer from 'react-test-renderer';

describe('CardComponent', () => {
    const mockUser: IUserInterface = {
        gender: "female",
        name: {
            title: "Mrs",
            first: "Perpétua",
            last: "Gomes"
        },
        location: {
            street: {
                number: 281,
                name: "Rua Amazonas "
            },
            city: "Araras",
            state: "Goiás",
            country: "Brazil",
            postcode: 21660,
            coordinates: {
                latitude: "-74.6426",
                longitude: "173.3337"
            },
            timezone: {
                offset: "-9:00",
                description: "Alaska"
            }
        },
        email: "perpetua.gomes@example.com",
        login: {
            uuid: "c7510105-283f-45bb-8d72-c78140925547",
            username: "blackrabbit543",
            password: "iceman1",
            salt: "Hn8wQTrZ",
            md5: "c86397fe886a0fcd63dff94dd02b2540",
            sha1: "47a2e1c8a869250f88997a2df0614f61a8767547",
            sha256: "9b9ee780cbf9f5832f625b35690449076156ecfb28e90c5c101b7ef561165f65"
        },
        dob: {
            date: "1973-05-07T08:44:57.810Z",
            age: 51
        },
        registered: {
            date: "2003-07-16T16:29:51.731Z",
            age: 20
        },
        phone: "(05) 1333-4691",
        cell: "(03) 2876-7794",
        id: {
            name: "CPF",
            value: "226.682.609-02"
        },
        picture: {
            large: "https://randomuser.me/api/portraits/women/25.jpg",
            medium: "https://randomuser.me/api/portraits/med/women/25.jpg",
            thumbnail: "https://randomuser.me/api/portraits/thumb/women/25.jpg"
        },
        nat: "BR"
    }

    it('renders correctly', () => {
        const tree = renderer.create(<CardComponent user={mockUser} onPress={() => { }} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('calls onPress when TouchableOpacity is pressed', () => {
        const onPressMock = jest.fn();
        const component = renderer.create(<CardComponent user={mockUser} onPress={onPressMock} />);
        const instance = component.root;

        instance.findByProps({ testID: 'card-component' }).props.onPress();

        expect(onPressMock).toHaveBeenCalledTimes(1);
    });
});
