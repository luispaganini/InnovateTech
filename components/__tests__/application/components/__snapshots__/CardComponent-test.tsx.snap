// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="card-component"
>
  <View
    colorTheme="light"
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#fff",
          "borderBottomLeftRadius": 2,
          "borderBottomRightRadius": 2,
          "borderColor": "#687076",
          "borderTopLeftRadius": 2,
          "borderTopRightRadius": 2,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "space-between",
          "marginBottom": 10,
          "overflow": "hidden",
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": "90%",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgba(103, 80, 164, 1)",
            "borderRadius": 75,
            "height": 150,
            "width": 150,
          },
          undefined,
        ]
      }
    >
      <Image
        accessibilityIgnoresInvertColors={true}
        source={
          {
            "uri": "https://randomuser.me/api/portraits/women/25.jpg",
          }
        }
        style={
          {
            "borderRadius": 75,
            "height": 150,
            "width": 150,
          }
        }
      />
    </View>
    <View
      style={
        [
          {
            "backgroundColor": "#fff",
          },
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "height": "100%",
            "justifyContent": "space-between",
            "marginLeft": 10,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#11181C",
            },
            undefined,
            undefined,
            {
              "fontSize": 16,
              "fontWeight": "600",
              "lineHeight": 24,
            },
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        Mrs Perp√©tua Gomes
      </Text>
      <View
        style={
          [
            {
              "backgroundColor": "#fff",
            },
            {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          female
        </Text>
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ]
          }
        >
          07/05/1973
        </Text>
      </View>
    </View>
  </View>
</View>
`;
